### PR TITLE
Added counter to prevent docker name collisions

### LINKED ISSUE #6: Static container name stops other language servers from starting
**Description**
The docker container is started with a static name.  This prevents more than one container running.

**Solution**
Change `lsp-docker-launch-new-container` to inject a random number into the container name:
```
(defun lsp-docker-launch-new-container (docker-container-name path-mappings docker-image-id server-command)
  (split-string
   (--doto (format "docker run --name %s --rm -i %s %s %s"
                   (format "%s-%d" docker-container-name (random 1000))
                   (->> path-mappings
                        (-map (-lambda ((path . docker-path))
                                (format "-v %s:%s" path docker-path)))
                        (s-join " "))
                   docker-image-id
                   server-command))
   " "))
```

### PR DESCRIPTION
**Pull request by rnikoopour**
Implementation to fix #6

### CODE REVIEW

File: lsp-docker.el [hunk 1]
```diff
@@ -45,7 +45,7 @@
 (defun lsp-docker-launch-new-container (docker-container-name path-mappings docker-image-id server-command)
   (split-string
    (--doto (format "docker run --name %s --rm -i %s %s %s"
-                   docker-container-name
+                   (format "%s-%d" docker-container-name (random 1000))
```
Comment (yyoncho) [inline lsp-docker.el]: `(random 1000)` does not guarantee that it will be unique - you may declare an index variable and increment it.

Comment (rnikoopour) [reply lsp-docker.el]: Added name via closure and increment it accordingly.

File: lsp-docker.el [hunk 2]
```diff
@@ -42,21 +42,27 @@
...
+		       (->> path-mappings
+			    (-map (-lambda ((path . docker-path))
+				    (format "-v %s:%s" path docker-path)))
+			    (s-join " "))
+		       docker-image-id
+		       server-command))
+       " ")))
+
+  (defun lsp-docker-exec-in-container (docker-container-name path-mappings docker-image-id server-command)
+    (progn
```
Comment (yyoncho) [inline lsp-docker.el]: no need from progn.

File: lsp-docker.el [hunk 3]
```diff
@@ -42,21 +42,27 @@
...
-                   docker-image-id
-                   server-command))
-   " "))
-
-(defun lsp-docker-exec-in-container (docker-container-name path-mappings docker-image-id server-command)
-  (split-string
-   (format "docker exec -i %s %s" docker-container-name server-command)))
+(let ((lsp-docker-name-suffix 0))
+  (defun lsp-docker-launch-new-container (docker-container-name path-mappings docker-image-id server-command)
+    (progn
```
Comment (yyoncho) [inline lsp-docker.el]: no need from progn.

File: lsp-docker.el [hunk 4]
```diff
@@ -42,21 +42,27 @@
...
+      (setq lsp-docker-name-suffix (+ 1 lsp-docker-name-suffix))
```
Comment (yyoncho) [inline lsp-docker.el]: nit: you may use `cl-incf`

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**